File: server/models/Note.js
const mongoose = require('mongoose');

const { Schema } = mongoose;

const schema = new Schema({
  title: {
    type: String,
    required: true,
    trim: true,
  },
  content: {
    type: String,
  },
  tags: [
    {
      type: String,
    },
  ],
  userId: {
    type: Schema.Types.ObjectId,
    required: true,
    ref: 'User',
  },
});

module.exports = mongoose.model('Note', schema);

File: server/models/note.rb
class Note
  include Mongoid::Document

  belongs_to :user
  validates :title, presence: true

  field :title, type: String
  field :content, type: String
  field :tags, type: Array

  def self.permitted_params(params)
    permitted = %w[_id title content user_id tags]
    params.select { |k, _| permitted.include?(k) }
  end
end
